jl: build failure due to ghc 9

### Describe the bug
```
builder for '/nix/store/rhspfhlw8414pfy394l77v9mp7fyqahv-jl-0.0.5.drv' failed with exit code 1;
last 10 log lines:
>    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> [7 of 9] Compiling JL.Inferer       ( src/JL/Inferer.hs, dist/build/JL/Inferer.o, dist/build/JL/Inferer.dyn_o )
>
> src/JL/Inferer.hs:12:1: warning: [-Wunused-imports]
>     The import of ‘Control.Monad.Writer’ is redundant
>       except perhaps to import instances from ‘Control.Monad.Writer’
>     To import instances alone, use: import Control.Monad.Writer()
>    |
> 12 | import           Control.Monad.Writer
>    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
For full logs, run 'nix log /nix/store/rhspfhlw8414pfy394l77v9mp7fyqahv-jl-0.0.5.drv'.
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. Build `pkgs.jl`

### Notify maintainers
@fgaz 
### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.1, NixOS, 22.05 (Quokka), 22.05.20220403.bc4b9ee`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0pre20220322_d5d4d98`
 - channels(root): `"nixos, nixpkgs"`
 - channels(anselmschueler): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Thank you for reporting this issue; I can reproduce it. The problem is not in the warning displayed in the last 10 log lines, but in an earlier error:

```
[4 of 9] Compiling JL.Serializer    ( src/JL/Serializer.hs, dist/build/JL/Serializer.o, dist/build/JL/Se
rializer.dyn_o )

src/JL/Serializer.hs:39:30: error:
    • Couldn't match type ‘Data.HashMap.Internal.HashMap Text’
                     with ‘Data.Aeson.KeyMap.KeyMap’
      Expected: Object
        Actual: Data.HashMap.Internal.HashMap Text Value
    • In the first argument of ‘Object’, namely ‘(fmap coreToValue hm)’
      In the expression: Object (fmap coreToValue hm)
      In a case alternative:
          RecordCore hm -> Object (fmap coreToValue hm)
   |     
39 |     RecordCore hm -> Object (fmap coreToValue hm)
   |                              ^^^^^^^^^^^^^^^^^^^
```

Unfortunately I don't think that I will be able to contribute further to this issue.

--- a/pkgs/development/tools/jl/default.nix
+++ b/pkgs/development/tools/jl/default.nix
@@ -5,21 +5,8 @@
 }:
 mkDerivation rec {
   pname = "jl";
-  version = "0.0.5";
-  src = fetchFromGitHub {
-    owner = "chrisdone";
-    repo = "jl";
-    rev = "v${version}";
-    sha256 = "1hlnwsl4cj0l4x8dxwda2fcnk789cwlphl9gv9cfrivl43mgkgar";
-  };
-  patches = [
-    # MonadFail compatibility patch. Should be removed with the next release
-    (fetchpatch {
-      url = "https://github.com/chrisdone/jl/commit/6d40308811cbc22a96b47ebe69ec308b4e9fd356.patch";
-      sha256 = "1pg92ffkg8kim5r8rz8js6fjqyjisg1266sf7p9jyxjgsskwpa4g";
-    })
-  ];
-
+  version = "0.1.0";
+  sha256 = "15vvn3swjpc5qmdng1fcd8m9nif4qnjmpmxc9hdw5cswzl055lkj";
   isLibrary = true;
   isExecutable = true;
   libraryHaskellDepends = [
@@ -34,10 +21,4 @@ mkDerivation rec {
   description = "Functional sed for JSON";
   maintainers = with lib.maintainers; [ fgaz ];
   homepage = "https://github.com/chrisdone/jl";
-
-  # jl needs to be updated to work with aeson-2.0.
-  # As far as I can tell, there is unfortunately no where to report issues upstream
-  # for us to be able to track when upstream adds aeson-2.0 support.
-  hydraPlatforms = lib.platforms.none;
-  broken = true;
 }

